minimint/src/net/connect.rs

```diff
@@ -97,13 +95,21 @@ where
         Ok(connections)
     }
 
-    async fn connect_to_peer(port: u16, peer: PeerId) -> Result<TcpStream, std::io::Error> {
+    async fn connect_to_peer(
+        port: u16,
+        peer: PeerId,
+        retries: u32,
+    ) -> Result<TcpStream, std::io::Error> {
         debug!("Connecting to peer {}", peer);
-        let res = TcpStream::connect(("127.0.0.1", port)).await;
-        if res.is_err() {
-            error!("Could not connect to peer {}", peer);
+        let mut counter = 0;
+        loop {
+            let res = TcpStream::connect(("127.0.0.1", port)).await;
+            if res.is_ok() || counter > retries {
```

Comment:
I think is off-by-one and should be `counter >= retries`

Reply:
I'll fix that.